govcd/media_test.go

```diff
@@ -196,3 +239,232 @@ func (vcd *TestVCD) Test_RefreshMediaImage(check *C) {
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+
+	itemName := "TestRefreshMedia"
+	uploadTask, err := catalog.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	AddToCleanupList(itemName, "mediaCatalogImage", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_RefreshMediaImage")
+
+	media, err := catalog.GetMediaByName(itemName, true)
+	check.Assert(err, IsNil)
+	check.Assert(media, NotNil)
+	check.Assert(media, Not(Equals), Media{})
+
+	oldMedia := media
+	err = media.Refresh()
+	check.Assert(err, IsNil)
+
+	check.Assert(media, NotNil)
+	check.Assert(oldMedia.Media.ID, Equals, media.Media.ID)
+	check.Assert(oldMedia.Media.Name, Equals, media.Media.Name)
+	check.Assert(oldMedia.Media.HREF, Equals, media.Media.HREF)
+}
+
+// Tests Media retrieval by name, by ID, and by a combination of name and ID
+func (vcd *TestVCD) Test_GetMedia(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_GetMedia: Org name not given")
+		return
+	}
+	if vcd.config.VCD.Catalog.Name == "" {
+		check.Skip("Test_GetMedia: Catalog name not given")
+		return
+	}
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+
+	itemName := "TestGetMedia"
+	uploadTask, err := catalog.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	AddToCleanupList(itemName, "mediaCatalogImage", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_GetMedia")
+
+	getByName := func(name string, refresh bool) (genericEntity, error) {
+		err = catalog.Refresh()
+		check.Assert(err, IsNil)
+		return catalog.GetMediaByName(name, refresh)
+	}
+	getById := func(id string, refresh bool) (genericEntity, error) {
+		return catalog.GetMediaById(id, refresh)
+	}
+	getByNameOrId := func(id string, refresh bool) (genericEntity, error) {
+		return catalog.GetMediaByNameOrId(id, refresh)
+	}
+
+	var def = getterTestDefinition{
+		parentType:    "Catalog",
+		parentName:    vcd.config.VCD.Catalog.Name,
+		entityType:    "Media",
+		entityName:    itemName,
+		getByName:     getByName,
+		getById:       getById,
+		getByNameOrId: getByNameOrId,
+	}
+	vcd.testFinderGetGenericEntity(def, check)
+}
+
+// Tests System function Delete by creating media and deleting it after.
+func (vcd *TestVCD) Test_DeleteMediaImage(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	skipWhenMediaPathMissing(vcd, check)
+	itemName := `Test_DeleteMediaImage`
+
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_GetMedia: Org name not given")
```

Comment:
`Test_GetMedia` should be `TestDeleteMediaImage`

Reply:
fixed